prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -319,4 +355,97 @@ object CustomFHIRFunctions {
         }
         return if (type != null) mutableListOf(StringType(type)) else mutableListOf()
     }
+
+    /**
+     * Get the LOINC Code from the LIVD table based on the device id, equipment model id, test kit name id, or the
+     * element model name
+     * @return a list with one value denoting the LOINC Code, or an empty list
+     */
+    fun livdTableLookup(
+        focus: MutableList<Base>,
+        parameters: MutableList<MutableList<Base>>?,
+        metadata: Metadata = Metadata.getInstance()
+    ): MutableList<Base> {
+        val lookupTable = metadata.findLookupTable(name = LivdLookup.livdTableName)
+
+        if (focus.size != 1) {
+            error("Must call the livdTableLookup function on a single observation")
+        }
+
+        val observation = focus.first()
+        if (observation !is Observation) {
+            error("Must call the livdTableLookup function on an observation")
```

Comment:
This cannot be error. make this a SchemaException, which is what is caught in the rest of the code.